### PR TITLE
[test-only] check secure view permissions in the Report response

### PR DESCRIPTION
**Pull request by ScharfViktor**
test for https://github.com/owncloud/ocis/issues/9607

### CODE REVIEW

File: tests/acceptance/features/apiContract/sharesReport.feature [hunk 1]
```diff
@@ -82,3 +82,66 @@ Feature: REPORT request to Shares space
...
+      | resource        | secureFolder  |
+      | space           | Personal      |
+      | sharee          | Brian         |
+      | shareType       | user          |
+      | permissionsRole | Secure viewer |
+    When user "Brian" searches for "secureFolder" using the WebDAV API
+    Then the HTTP status code should be "207"
+    # And the following headers should match these regular expressions
+    #   | X-Request-Id | /^[a-zA-Z]+\/[a-zA-Z]+\.feature:\d+(-\d+)?$/ |
+    And the "REPORT" response to user "Brian" should contain a mountpoint "secureFolder" with these key and value pairs:
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/sharesReport.feature:101]: do we need to check the request-id header?

File: tests/acceptance/features/apiContract/sharesReport.feature [hunk 2]
```diff
@@ -82,3 +82,66 @@ Feature: REPORT request to Shares space
...
+    And user "Alice" has sent the following resource share invitation:
+      | resource        | secure.txt    |
+      | space           | Personal      |
+      | sharee          | Brian         |
+      | shareType       | user          |
+      | permissionsRole | Secure viewer |
+    When user "Brian" searches for "secure.txt" using the WebDAV API
+    Then the HTTP status code should be "207"
+    And the following headers should match these regular expressions
+      | X-Request-Id | /^[a-zA-Z]+\/[a-zA-Z]+\.feature:\d+(-\d+)?$/ |
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/sharesReport.feature:136]: do we need to check the request-id header?

File: tests/acceptance/features/apiContract/sharesReport.feature [hunk 3]
```diff
@@ -82,3 +82,66 @@ Feature: REPORT request to Shares space
       | dav-path-version |
       | old              |
       | new              |
+
+
+  Scenario Outline: check the REPORT response of the found security folder
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/sharesReport.feature:87]:
```suggestion
  Scenario Outline:  check the REPORT response of a folder shared with secure viewer role
```

File: tests/acceptance/features/apiContract/sharesReport.feature [hunk 4]
```diff
@@ -82,3 +82,66 @@ Feature: REPORT request to Shares space
...
+      | d:getcontenttype   | text/plain    |
+      | oc:permissions     | SX            |
+      | d:getcontentlength | 14            |
+    Examples:
+      | dav-path-version |
+      | old              |
+      | new              |
+
+
+  Scenario Outline: check the REPORT response of the found security file
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/sharesReport.feature:124]:
```suggestion
  Scenario Outline: check the REPORT response of a file shared with secure viewer role
```

File: tests/acceptance/features/apiContract/sharesReport.feature [hunk 5]
```diff
@@ -82,3 +82,66 @@ Feature: REPORT request to Shares space
...
+      | oc:shareroot       | /secureFolder |
+      | oc:name            | secure.txt    |
+      | d:getcontenttype   | text/plain    |
+      | oc:permissions     | SX            |
+      | d:getcontentlength | 14            |
+    Examples:
+      | dav-path-version |
+      | old              |
+      | new              |
+
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/sharesReport.feature:122]: if it is possible to check with spaces endpoint as well.
```suggestion
      | old              |
      | new              |
      | spaces           |

```

File: tests/acceptance/features/apiContract/sharesReport.feature (hunk 1 above)
Comment (ScharfViktor) [reply tests/acceptance/features/apiContract/sharesReport.feature:101]: I forgot to enable it

File: tests/acceptance/features/apiContract/sharesReport.feature [hunk 6]
```diff
@@ -82,3 +82,70 @@ Feature: REPORT request to Shares space
       | dav-path-version |
       | old              |
       | new              |
+
+
+  Scenario Outline: check the REPORT response of a folder shared with secure viewer role
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/sharesReport.feature:87]:
```suggestion
  @issue-9607
  Scenario Outline: check the REPORT response of a folder shared with secure viewer role
```

File: tests/acceptance/features/apiContract/sharesReport.feature [hunk 7]
```diff
@@ -82,3 +82,70 @@ Feature: REPORT request to Shares space
...
+      | oc:permissions     | SX            |
+      | d:getcontentlength | 14            |
+    Examples:
+      | dav-path-version |
+      | old              |
+      | new              |
+      | spaces           |
+
+
+  Scenario Outline: check the REPORT response of a file shared with secure viewer role
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/sharesReport.feature:127]:
```suggestion
  @issue-9607
  Scenario Outline: check the REPORT response of a file shared with secure viewer role
```

**Review by saw-jan — APPROVED** [review]
can we add issue tag for future reference? others are good

File: createFiles.sh [hunk 8]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+# for i in {1..100}
+# do
+#   response=$(curl -s -o /dev/null -w "%{http_code}" -X PUT "https://localhost:9200/remote.php/dav/spaces/4811f8a3-75fe-4cb1-b1d0-decb3588c927%249f10807c-b1c6-4e2e-9bdc-7f3f8ac5fb3a/${i}.txt" -u admin:admin --data "This is the content of file ${i}.txt")
```
Comment (saw-jan) [inline createFiles.sh:4]: by mistake :smile:

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: createFiles.sh (hunk 8 above)
Comment (ScharfViktor) [reply createFiles.sh]: opss. deleted

**Review by saw-jan — APPROVED** [review]
:+1:

**Pull request merged**